Intersect.Editor/Forms/DockingElements/frmMapEditor.cs

```diff
@@ -2343,6 +2366,115 @@ private void picMap_Paint(object sender, PaintEventArgs e)
+
+            if (!(!string.IsNullOrEmpty(enableCursorSprites) && Convert.ToBoolean(enableCursorSprites)) ||
+                string.IsNullOrEmpty(mCurPath) ||
+                !File.Exists(mCurPath))
+            {
+                return;
+            }
+
+            mCurSprite.Dispose();
+            DestroyIcon(Cursor.Handle);
+            Cursor = Cursors.Default;
+        }
+
+        private void picMap_KeyDown(object sender, KeyEventArgs e)
+        {
+            var toolKeyIsPressed = (e.KeyData == Keys.B || e.KeyData == Keys.M || e.KeyData == Keys.R ||
+                                    e.KeyData == Keys.F || e.KeyData == Keys.E || e.KeyData == Keys.I);
+            if (toolKeyIsPressed)
+            {
+                SetCursorSpriteInGrid();
+            }
+        }
+
+        private void SetCursorSpriteInGrid()
+        {
+            var enableCursorSprites = Preferences.LoadPreference("EnableCursorSprites");
+
+            if (!(!string.IsNullOrEmpty(enableCursorSprites) && Convert.ToBoolean(enableCursorSprites)) ||
+                Globals.CurrentEditor != -1)
+            {
+                return;
+            }
+
+            picMap.Focus();
+
+            switch (Globals.CurrentTool)
+            {
+                case (int)EditingTool.Brush:
+                    mCurPath = $"resources/cursors/editor_{EditingTool.Brush.ToString().ToLowerInvariant()}.png";
+                    mCurClickPoint = ToolCursor.Brush.CursorClickPoint;
+                    break;
+
+                case (int)EditingTool.Selection:
+                    mCurPath = $"resources/cursors/editor_{EditingTool.Selection.ToString().ToLowerInvariant()}.png";
+                    mCurClickPoint = ToolCursor.Selection.CursorClickPoint;
+                    break;
+
+                case (int)EditingTool.Rectangle:
+                    mCurPath = $"resources/cursors/editor_{EditingTool.Rectangle.ToString().ToLowerInvariant()}.png";
+                    mCurClickPoint = ToolCursor.Rectangle.CursorClickPoint;
+                    break;
+
+                case (int)EditingTool.Fill:
+                    mCurPath = $"resources/cursors/editor_{EditingTool.Fill.ToString().ToLowerInvariant()}.png";
+                    mCurClickPoint = ToolCursor.Fill.CursorClickPoint;
+                    break;
+
+                case (int)EditingTool.Erase:
+                    mCurPath = $"resources/cursors/editor_{EditingTool.Erase.ToString().ToLowerInvariant()}.png";
+                    mCurClickPoint = ToolCursor.Erase.CursorClickPoint;
+                    break;
+
+                case (int)EditingTool.Dropper:
+                    mCurPath = $"resources/cursors/editor_{EditingTool.Dropper.ToString().ToLowerInvariant()}.png";
+                    mCurClickPoint = ToolCursor.Dropper.CursorClickPoint;
+                    break;
+
+                default:
+                    return;
+            }
```

Comment:
I said this switch didn't need to exist anymore. Using a dictionary this can be reduced to 2 assignment lines

Reply:
```
mCurPath = $"resources/cursors/editor_{Globals.CurrentTool.ToString().ToLowerInvariant()}.png";
mCurClickPoint = ToolCursor.ToolCursorDict[(EditingTool)Globals.CurrentTool].CursorClickPoint;
```